### PR TITLE
chore: warn on fuel client version incompatibility

### LINKED ISSUE #2207: Use soft warnings for node incompatible versions
> Related https://github.com/FuelLabs/fuels-rs/pull/1348.

Consider printing warnings instead of throwing for `major` and `minor`:

https://github.com/FuelLabs/fuels-ts/blob/cfc4fde37a2f2d58a4c97d81dfe9f00756068669/packages/account/src/providers/provider.ts#L442-L447

### PR DESCRIPTION
**Pull request by nedsalk**
Similar to what was done on fuels-rs [here](https://github.com/FuelLabs/fuels-rs/pull/1348).

closes #2207

### CODE REVIEW

**Comment by fuel-service-user** [comment]
✨ A PR has been created under the `rc-2209` tag on `fuels-wallet` repo.
https://github.com/FuelLabs/fuels-wallet/pull/1260

**Review by arboleya — COMMENTED** [review]
Although the behavior changes a bit, perhaps it could be a patch since it doesn't break the API.

The changes to `error-codes.md` should probably be left out.

File: .changeset/soft-knives-sit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/account": minor
+---
+
+chore!: warn on fuel client version incompatibility
```
[the remark this answers is not in the document]
Comment (nedsalk) [reply .changeset/soft-knives-sit.md:5]: I looked at it from the viewpoint of an error not being thrown anymore (public behavior) so I set it to being a breaking minor version.

File: packages/account/src/providers/provider.ts [hunk 2]
```diff
@@ -440,9 +440,9 @@ export default class Provider {
       checkFuelCoreVersionCompatibility(nodeInfo.nodeVersion);
 
     if (!isMajorSupported || !isMinorSupported) {
-      throw new FuelError(
-        FuelError.CODES.UNSUPPORTED_FUEL_CLIENT_VERSION,
-        `Fuel client version: ${nodeInfo.nodeVersion}, Supported version: ${supportedVersion}`
+      // eslint-disable-next-line no-console
+      console.warn(
```
[the remark this answers is not in the document]
Comment (nedsalk) [reply packages/account/src/providers/provider.ts:444]: It was tried [previously](https://github.com/FuelLabs/fuels-ts/pull/1855#discussion_r1519610036). I just tried @danielbate's proposal from [here](https://github.com/FuelLabs/fuels-ts/pull/1855#discussion_r1521155742) but it didn't work.

File: .changeset/soft-knives-sit.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nedsalk) [reply .changeset/soft-knives-sit.md]: https://github.com/FuelLabs/fuels-ts/pull/2209/commits/d3ae81025a0948cba55c2526b20c3fe5ac71af4e

File: packages/account/src/providers/provider.ts [hunk 3]
```diff
@@ -440,9 +440,9 @@ export default class Provider {
...
+        `Unsupported fuel client version: ${nodeInfo.nodeVersion}, Supported version: ${supportedVersion}`
```
[the remark this answers is not in the document]
Comment (nedsalk) [reply packages/account/src/providers/provider.ts]: https://github.com/FuelLabs/fuels-ts/pull/2209/commits/1f0a59d106bd82d3bd5cb593a560e7f4d9b5c472

**Review by arboleya — APPROVED** [review]

**Review by maschad — APPROVED** [review]

File: packages/account/src/providers/provider.ts [hunk 4]
```diff
@@ -441,9 +441,9 @@ export default class Provider {
       checkFuelCoreVersionCompatibility(nodeInfo.nodeVersion);
 
     if (!isMajorSupported || !isMinorSupported) {
-      throw new FuelError(
-        FuelError.CODES.UNSUPPORTED_FUEL_CLIENT_VERSION,
-        `Fuel client version: ${nodeInfo.nodeVersion}, Supported version: ${supportedVersion}`
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Unsupported Fuel client version: ${nodeInfo.nodeVersion}, Supported version: ${supportedVersion}`
```
Comment (Dhaiwat10) [inline packages/account/src/providers/provider.ts:446]: Maybe we could improve the warning by replacing it with something like:

> The Fuel Node that you are trying to connect to is using a `fuel-core` version that is incompatible with the version of the TS SDK that you are using.

**Review by Dhaiwat10 — DISMISSED** [review]

**Review by danielbate — APPROVED** [review]

File: packages/account/src/providers/provider.ts (hunk 4 above)
Comment (nedsalk) [reply packages/account/src/providers/provider.ts]: https://github.com/FuelLabs/fuels-ts/pull/2209/commits/49c48fa9742daf1ef3e1ebd2df4852b3761834d1

**Review by danielbate — DISMISSED** [review]

**Review by arboleya — DISMISSED** [review]

File: .changeset/soft-knives-sit.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/account": minor
```
[the remark this answers is not in the document]
Comment (arboleya) [reply .changeset/soft-knives-sit.md:2]: @petertonysmith94 Good catch!

**Review by arboleya — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]

File: packages/account/src/providers/provider.test.ts [hunk 6]
```diff
@@ -862,13 +862,20 @@ describe('Provider', () => {
...
-    });
+    const consoleWarnSpy = vi.spyOn(console, 'warn');
+
+    await Provider.create(FUEL_NETWORK_URL);
+
+    expect(consoleWarnSpy).toHaveBeenCalledOnce();
+    expect(consoleWarnSpy).toHaveBeenCalledWith(
+      `The Fuel Node that you are trying to connect to is using fuel-core version ${FUEL_CORE}, 
+which is not supported by the version of the TS SDK that you are using. 
+Things might or might not work as expected.
```
Comment (maschad) [inline packages/account/src/providers/provider.test.ts:873]:
```suggestion
Things may not work as expected.
```

File: packages/account/src/providers/provider.test.ts [hunk 7]
```diff
@@ -887,10 +894,17 @@ describe('Provider', () => {
...
```
Comment (maschad) [inline packages/account/src/providers/provider.test.ts:905]:
```suggestion
Things may not work as expected.
```

File: packages/account/src/providers/provider.ts [hunk 8]
```diff
@@ -441,9 +441,12 @@ export default class Provider {
...
 
     if (!isMajorSupported || !isMinorSupported) {
-      throw new FuelError(
-        FuelError.CODES.UNSUPPORTED_FUEL_CLIENT_VERSION,
-        `Fuel client version: ${nodeInfo.nodeVersion}, Supported version: ${supportedVersion}`
+      // eslint-disable-next-line no-console
+      console.warn(
+        `The Fuel Node that you are trying to connect to is using fuel-core version ${nodeInfo.nodeVersion}, 
+which is not supported by the version of the TS SDK that you are using. 
+Things might or might not work as expected.
```
Comment (maschad) [inline packages/account/src/providers/provider.ts:448]:
```suggestion
Things may not work as expected.
```

**Review by maschad — CHANGES_REQUESTED** [review]
Grammar pedant 🤓

**Review by arboleya — APPROVED** [review]

**Review by maschad — APPROVED** [review]
LGTM

**Pull request merged**